src/internal/executer.ts

```diff
@@ -21,7 +21,7 @@ export class RequestExecuter {
     await this.systemInit;
     const token = await this.tokenManager.token(this.config.auth);
     return this.requestAdapter.execute(
-      this.getUrl(),
+      this.getURI(request.queryParams || []),
```

Comment:
I think you already set an empty array if there are no params in BaseQuery:
```
queryParams: this.query.compileToQueryParams() || [],
```
do you need to do it again here?

Reply:
I don't, I was just afraid of committing mistakes :laughing: 
If I just remove it typescript complains about the type `getURI` expects, so I removed ` || []` and changed method signature to accept empty array by default
```js
private getURI(queryParams: QueryParam[] = [])
```